lib/execution_segment.go

```diff
@@ -730,3 +730,79 @@ func (et *ExecutionTuple) GetNewExecutionTupleFromValue(value int64) (*Execution
 		SegmentIndex: newIndex,
 	}, nil
 }
+
+// SegmentedIndex ...
+type SegmentedIndex struct {
+	start, lcd       int64
+	offsets          []int64
+	scaled, unscaled int64 // for both the first element(vu) is 1 not 0
+}
+
+// NewSegmentedIndex returns a pointer to a new SegmentedIndex instance,
+// given a starting index, LCD and offsets as returned by GetStripedOffsets().
+func NewSegmentedIndex(start, lcd int64, offsets []int64) *SegmentedIndex {
```

Comment:
btw, writing the comment suggestion above, we can probably make this constructor accept the `ExecutionTuple` directly. As far as I can see, this is the only way it's always going to be called, so we can just have it call `et.GetStripedOffsets()` internally?

Reply:
Done in ed07691a.